working/macros/feature-specification.md

```diff
@@ -393,6 +393,115 @@ be constructed, but where you need to build up something in smaller pieces you
 can use the [Fragment][] subtype. Any arbitrary String can be passed to this
 class, allowing you to build up your code fragments however you like.
 
+### Resource API
+
+Some macros may wish to load resources (such as files). We do want to enable
```

Comment:
Devil's advocate: Do we really? This isn't needed by any of our core use cases.

If we go down the road of having macros compile to something file-based—essentially a sort of layered approach to static metaprogramming—then I think it's reasonable to say that metaprogramming that needs external files shouldn't use macros but could still use partial classes/augmentations/whatever.

Reply:
We chatted a bit offline here but just for posterity - the main use case here is for any sort of framework that wants to expose a template language, and allow those templates to live in a separate file.

That covers not only a bunch of potential UI frameworks but also even server side rendering frameworks that want to pre-compile their template files with a macro. I think even just that one general category of use cases is pretty compelling. I am not tied specifically to this API though.

Reply:
FWIW, I'm quite interested in the use cases this would enable. Specifically, asset management capabilities for flutter apps.

Reply:
Yes this could also be used for some asset bundle related purpose, although you would want to be careful doing so. If the assets are large you could end up adding a fair bit of build time to your app on accident.